testing/steps/test_zip_data.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+
+from neuraxle.base import ExecutionContext
+from neuraxle.data_container import DataContainer
+from neuraxle.pipeline import Pipeline
+from neuraxle.steps.data import ZipData
+
+TIMESTEPS = 10
+FEATURES = 5
+VALIDATION_SIZE = 0.1
+BATCH_SIZE = 32
+N_EPOCHS = 10
+SHAPE_3D = (BATCH_SIZE, TIMESTEPS, FEATURES)
+SHAPE_2D = (BATCH_SIZE, TIMESTEPS)
+SHAPE_1D = BATCH_SIZE
+
+
+def test_zip_data_should_merge_3d_with_3d():
+    # Given
+    data_inputs_3d, expected_outputs_3d = _create_data_source(SHAPE_3D)
+    data_inputs_3d_second, expected_outputs_3d_second = _create_data_source(SHAPE_3D)
+    data_container_3d_second = DataContainer(data_inputs=data_inputs_3d_second, expected_outputs=expected_outputs_3d_second)
+    data_container = DataContainer(data_inputs=data_inputs_3d, expected_outputs=expected_outputs_3d) \
+        .add_sub_data_container('2d', data_container_3d_second)
+
+    # When
+    p = Pipeline([
+        ZipData(data_sources=['2d'])
```

Comment:
3D in given, 2D in when. I don't understand what is happening

Reply:
Note: I didn't think it was really a string here. Why is it a string? How does it work exactly?

Reply:
@guillaume-chevalier have you read the description of the pr ? it should help you understand what I did here. the naming is probably off. trying to find a better name for the data_sources parameter